### ISSUE TITLE
Non English "Encoding" test

### ISSUE THREAD

**Issue by zigm** [tomwillow/SmartCharsetConverter#14]
My language is sett to English but Encoding and Line Breaks are in Chinese ?? I am not able to resolve that.

![image](https://github.com/user-attachments/assets/5b38742e-e9ed-4e32-989e-a88d40a3d73f)

**Comment by tomwillow** [comment]
This issue has been confirmed.
The tool was initially developed for my personal use, so some areas may not be well localized.
Once the issue is fixed, I will notify you.

Thank you for your feedback!

P.S. "未知" means "Unknown".

**Comment by zigm** [comment]
Please find attached the file i tried to verify. I think the encoding should be Western European.
[italian.txt](https://github.com/user-attachments/files/18141303/italian.txt)

**Comment by zigm** [comment]
I am trying to build the tool on windows 10. Installed the latest release and vcpkg and attempted to run config_on_win.bat.
I am getting an error. Hope I am missing some steps.
Thank You
```

C:\Users\tata\Downloads\SmartCharsetConverter-0.9.2\SmartCharsetConverter-0.9.2>echo %VCPKG_ROOT%
**C:\path\to\vcpkg**

C:\Users\tata\Downloads\SmartCharsetConverter-0.9.2\SmartCharsetConverter-0.9.2>config_on_win.bat

C:\Users\tata\Downloads\SmartCharsetConverter-0.9.2\SmartCharsetConverter-0.9.2>cmake --preset=default
CMake Error: Could not read presets from C:/Users/tata/Downloads/SmartCharsetConverter-0.9.2/SmartCharsetConverter-0.9.2: **Unrecognized "version" field**

C:\Users\tata\Downloads\SmartCharsetConverter-0.9.2\SmartCharsetConverter-0.9.2>pause

```

**Comment by zigm** [comment]
VCPKG was not set properly. Corrected but error stays the same.

```

C:\Users\tata>cd vcpkg

C:\Users\tata\vcpkg>bootstrap-vcpkg.bat
Downloading https://github.com/microsoft/vcpkg-tool/releases/download/2024-12-09/vcpkg.exe -> C:\Users\tata\vcpkg\vcpkg.exe... done.
Validating signature... done.

vcpkg package management program version 2024-12-09-1005b78fa1bf1dde1a20c2734cba4ea61ca94d9a


C:\Users\tata\Downloads\SmartCharsetConverter-0.9.2\SmartCharsetConverter-0.9.2>echo %VCPKG_ROOT%
C:\Users\tata\vcpkg


```

**Comment by zigm** [comment]
Upgraded cmake to make config_on_win.bat work. Didn't realize this will result in quite large download, etc. 

./build/SmartCharsetConverter.sln. was created. Tried to build under vs2022 but there seems to be one issue. Will investigate.


```

Build started at 2:25 AM...
1>------ Build started: Project: uchardet, Configuration: Debug x64 ------
1>cl : command line  warning D9002: ignoring unknown option '-ffloat-store'
1>uchardet.cpp
1>C:\Users\tata\Downloads\SmartCharsetConverter-0.9.2\SmartCharsetConverter-0.9.2\third_party\uchardet\uchardet\src\tools\uchardet.cpp(37,10): warning C4464: relative include path contains '..'
1>C:\Users\tata\Downloads\SmartCharsetConverter-0.9.2\SmartCharsetConverter-0.9.2\third_party\uchardet\uchardet\src\tools\uchardet.cpp(41,10): error C1083: Cannot open include file: 'getopt.h': No such file or directory
1>Done building project "uchardet.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 9 up-to-date, 0 skipped ==========
========== Build completed at 2:25 AM and took 01.079 seconds ==========

```

**Comment by zigm** [comment]
I installed getopt.h from web and was able to compile all components.

Testing with the file I provided results in the following error message. Notepad++ says file encoding is ANSI. Using notepad++ I can convert the file to UTF16LE and UTF16BE and your program detect encoding correctly. It doesn't seem to like ANSI encoded version.

![image](https://github.com/user-attachments/assets/1de1115f-5c5f-4f64-9f3b-2acd4271d7ff)

**Comment by tomwillow** [comment]
I opened this file by VSCode with Hex Editor extension:

![image](https://github.com/user-attachments/assets/85f0ebaa-03ef-4aca-a76c-a4c61c7ad7d1)

This snapshot shows that its original encoding has been lost. Notice the "3F" is the ASCII character "?", this phenomenon occurred while a text is decoded by wrong encoding, and then copied it (this copy operation makes key encoding infomation lost!)

So, this file can't be detected anymore, its "?" is permanent and not means "unresolved character".

This is why Notepad++ report it as ANSI (ANSI is actually your machine's LOCAL ENCODING. For example, in my machine, ANSI means GB18030(a Chinese charset)).

**Comment by tomwillow** [comment]
> Please find attached the file i tried to verify. I think the encoding should be Western European. [italian.txt](https://github.com/user-attachments/files/18141303/italian.txt)

If you open it with "Windows-1252""ISO-8859-1"(two normal west europe charset), the shown content of it will not change (especially the "?" character).

**Comment by zigm** [comment]
Thank you for detailed analysis. You are right that the file looks corrupted.

I am prototyping support for multiple languages by mbox mail viewer (open source MBox Viewer) and created few translation files using Google Translator. I merged two files, English version and translated into Italian in the example.

I was wondering about encoding done by Google Translator but was trying to check encoding of the merged file instead of just translated text. I checked now just the translated text from Google and it is reported as UTF-8 as expected.  My merging of two files broke encoding somehow and I will need to investigate. I was looking for a tool to double check encoding.

Thanks for your help.

**Comment by zigm** [comment]
Thanks. I may try at some point with the suspect files. Translated web page to English using Google Translator ability to translate web link. Sounds I can start with the plain text or text in hex. Encoding list is limited but I may not  need the full list.

**Comment by tomwillow** [comment]
The Chinese word "未知" has been changed to "Unknown".

Fixed at v0.9.3: https://github.com/tomwillow/SmartCharsetConverter/releases/tag/v0.9.3
